flink-runtime-web/src/main/java/org/apache/flink/runtime/webmonitor/history/HistoryServerArchiveFetcher.java

```diff
@@ -258,7 +269,13 @@ private static String convertLegacyJobOverview(String legacyOverview) throws IOE
 		int[] tasksPerState = new int[ExecutionState.values().length];
 		// pending is a mix of CREATED/SCHEDULED/DEPLOYING
 		// to maintain the correct number of task states we have to pick one of them
-		tasksPerState[ExecutionState.SCHEDULED.ordinal()] = pending;
+		if (versionLessThan14) {
+			tasksPerState[ExecutionState.SCHEDULED.ordinal()] = pending;
+		} else {
+			tasksPerState[ExecutionState.SCHEDULED.ordinal()] = scheduled;
+			tasksPerState[ExecutionState.CREATED.ordinal()] = created;
+			tasksPerState[ExecutionState.DEPLOYING.ordinal()] = deploying;
+		}
```

Comment:
Do we need the `versionLessThan14` distinction here? We could say that we assign `scheduled = pendingNode.asInt()` in line 257 and get completely rid of `pending`.